go/mysql/query_test.go

```diff
@@ -1114,3 +1115,36 @@ func TestExecuteQueries(t *testing.T) {
 		}
 	})
 }
+
+func TestComPrepareMultiple(t *testing.T) {
```

Comment:
Maybe rename to `TestComStmtPrepareWithTrailingNewline` or something? That's a little more descriptive of what you're testing here (there aren't actually multiple statements, right?) and more consistent with the naming of the existing test `TestComStmtPrepare`. 

Speaking of multiple statements... do we have a test in here that tests the "prepare fails for multiple statements" case? It seems like that would be useful if we don't have it.